### ISSUE TITLE
Questions about ranker loss

### ISSUE THREAD

**Issue by ShengleiH** [cookielee77/RankGan-NIPS2017#2]
Hi @kevinlin311tw @cookielee77 
I have a question about the loss in ranker:

In line 157 in ranker.py: `self.loss = -(pos_loss - neg_loss)`
I am wondering why the loss here is negative. If it is a logloss, then it should be `self.loss = -(pos_loss + neg_loss)`

Looking forward to your reply, thank you~

**Comment by cookielee77** [comment]
Hi,

Actually, the loss is a ranking score. During adversarial training, the ranker wants correctly distinguishing real/fake sentences. Thus, it maximizes the scores of positive sentences (pos_loss), and minimizes the scores of negative sentences (neg_loss). That's why there is a negative sign before neg_loss. 

Hope this resolve your question.

**Comment by ShengleiH** [comment]
@cookielee77 Thank you for your reply! So this loss should tend to negative Infinity?

**Comment by cookielee77** [comment]
Ideally, if the neg score is zero, the neg_loss will be neg infinity. However, we didn't find this phenomenon in the adversarial experiment. If you are worried about the neg infinity, you can try the original GAN loss function, which avoids this neg infinity issue.

**Comment by ShengleiH** [comment]
@cookielee77 Thanks a lot!
